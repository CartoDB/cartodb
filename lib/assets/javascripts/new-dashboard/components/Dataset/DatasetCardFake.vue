<template>
  <div class="dataset-fake-row">
    <div class="cell-placeholder">
      <div class="icon-placeholder"/>
    </div>
    <div class="cell-placeholder cell--main">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--large">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--small">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--small">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--small">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--small">
      <div class="text-placeholder"/>
    </div>
    <div class="cell-placeholder cell--quick-actions">
    </div>
  </div>
</template>

<script>
export default {
  name: 'FakeDatasetCard'
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.dataset-fake-row {
  display: flex;
  align-items: center;
  width: 100%;
  height: 80px;
  padding: 22px 2px;
  background-color: $white;
}

.icon-placeholder {
  width: 36px;
  height: 36px;
  border-radius: 2px;
  background-color: $softblue;
}

.text-placeholder {
  height: 24px;
  background-color: $softblue;
}

.cell-placeholder {
  padding: 0 10px;
}

.cell--main {
  flex-grow: 1;
  flex-shrink: 1;
  min-width: 200px;
}

.cell--large {
  width: 165px;
}

.cell--medium {
  width: 120px;
}

.cell--small {
  width: 80px;
}

.cell--quick-actions {
  padding: 0 24px;
}
</style>
