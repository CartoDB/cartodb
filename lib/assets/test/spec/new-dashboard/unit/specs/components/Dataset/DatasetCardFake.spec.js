import { shallowMount } from '@vue/test-utils';
import DatasetCardFake from 'new-dashboard/components/Dataset/DatasetCardFake';

describe('DatasetCardFake.vue', () => {
  it('should render correct contents', () => {
    const datasetCardFake = shallowMount(DatasetCardFake);
    expect(datasetCardFake).toMatchSnapshot();
  });
});
