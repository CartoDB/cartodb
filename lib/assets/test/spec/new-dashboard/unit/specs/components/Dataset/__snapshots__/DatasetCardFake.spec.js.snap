// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetCardFake.vue should render correct contents 1`] = `
<div class="dataset-fake-row">
  <div class="cell-placeholder">
    <div class="icon-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--main">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--large">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--small">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--small">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--small">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--small">
    <div class="text-placeholder"></div>
  </div>
  <div class="cell-placeholder cell--quick-actions"></div>
</div>
`;
